src/common/src/FFmpegFilter.cpp

```diff
@@ -0,0 +1,79 @@
+#include <FFmpegFilter.h>
+#include <fmt/format.h>
+
+namespace FFmpegFilter {
+    /*
+     * Generate a string containing a "crop" filter that can be passed to FFmpeg so it will crop out overlapping regions
+     *
+     * Sample cfg = {
+     *   well_spacing = 9000;
+     *   xy_pixel_size = 41.67;
+     *   scale = 2;
+     *   num_wells_v = 4;
+     *   num_wells_h = 4;
+     *   frameWidth = 512;
+     *   frameHeight = 512;
+     *   rows = 2;
+     *   cols = 3;
+     *   v_offset = 0;
+     *   h_offset = 0;
+     * }
+     *
+     * Sample output for sample cfg (well_spacing_scaled = 108, so the 4x4 region preserved per FOV is 432x432):
+     *
+     * split=6[L_0_0][L_0_1][L_0_2][L_1_0][L_1_1][L_1_2];
+     * [L_0_0] crop=432:432:40:40 [L_0_0_c];
+     * [L_0_1] crop=432:432:552:40 [L_0_1_c];
+     * [L_0_2] crop=432:432:1064:40 [L_0_2_c];
+     * [L_1_0] crop=432:432:40:552 [L_1_0_c];
+     * [L_1_1] crop=432:432:552:552 [L_1_1_c];
+     * [L_1_2] crop=432:432:1064:552 [L_1_2_c];
+     * [L_0_0_c][L_0_1_c][L_0_2_c] hstack=inputs=3 [R0];
+     * [L_1_0_c][L_1_1_c][L_1_2_c] hstack=inputs=3 [R1];
+     * [R0][R1] vstack"
+     *
+     * See https://www.ffmpeg.org/ffmpeg-filters.html for more details
+     */
+    std::string getCropFilter(FFmpegFilterCfg* cfg) {
```

Comment:
The code for calculating the rois here looks the same as the `roiOffsets` code in Rois.cpp, I think it would be better to reuse the existing code instead of duplicating it here.

Reply:
Or add in the new functionality needed for creating the ffmpeg filter to the Rois namespace instead so all the logic dealing with rois is in one place.